src/util/templates/EjsTemplateEngine.ts

```diff
@@ -1,35 +1,25 @@
-/* eslint-disable tsdoc/syntax */
-// tsdoc/syntax cannot handle `@range`
-import type { TemplateFunction } from 'ejs';
-import { compile, render } from 'ejs';
-import type { TemplateEngine, Template } from './TemplateEngine';
-import { getTemplateFilePath, readTemplate } from './TemplateEngine';
+import { render } from 'ejs';
+import { ExtensionBasedTemplateEngine } from './ExtensionBasedTemplateEngine';
+import type { TemplateEngineInput } from './TemplateEngine';
+import { getTemplateFilePath, readTemplate } from './TemplateUtil';
 import Dict = NodeJS.Dict;
 
 /**
  * Fills in EJS templates.
  */
-export class EjsTemplateEngine<T extends Dict<any> = Dict<any>> implements TemplateEngine<T> {
-  private readonly applyTemplate: Promise<TemplateFunction>;
+export class EjsTemplateEngine<T extends Dict<any> = Dict<any>> extends ExtensionBasedTemplateEngine<T> {
   private readonly baseUrl: string;
 
   /**
    * @param baseUrl - Base URL of the server.
-   * @param template - The default template @range {json}
    */
-  public constructor(baseUrl: string, template?: Template) {
-    // EJS requires the `filename` parameter to be able to include partial templates
-    const filename = getTemplateFilePath(template);
+  public constructor(baseUrl: string) {
+    super([ 'ejs' ]);
     this.baseUrl = baseUrl;
-
-    this.applyTemplate = readTemplate(template)
-      .then((templateString: string): TemplateFunction => compile(templateString, { filename }));
```

Comment:
Would you happen to know why the compile step is no longer needed due to the changes? The registration page still renders correctly from what I see so partial templates still work. I just don't fully understand why since this was needed before.

Reply:
Because providing a default template via the constructor is no longer supported (and it was this template that needed compiling in the previous version). Due to the changes, the template is always provided via the call to `handle`, thus the execution path remains effectively unchanged.